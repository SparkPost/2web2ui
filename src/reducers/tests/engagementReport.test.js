import engagementReportReducer from '../engagementReport';
import cases from 'jest-in-case';

cases('Engagement Report Reducer', (action) => {
  expect(engagementReportReducer(undefined, action)).toMatchSnapshot();
}, {
  'when failed to get chart data': {
    type: 'GET_ENGAGEMENT_AGGREGATE_METRICS_FAIL',
    payload: {
      message: 'Oh no!'
    }
  },
  'when pending to get chart data': {
    type: 'GET_ENGAGEMENT_AGGREGATE_METRICS_PENDING'
  },
  'when succeeds to get chart data': {
    type: 'GET_ENGAGEMENT_AGGREGATE_METRICS_SUCCESS',
    payload: [{ count_accepted: 237433, count_targeted: 272234 }]
  },
  'when failed to get table data': {
    type: 'GET_ENGAGEMENT_LINK_METRICS_FAIL',
    payload: {
      message: 'Oh no!'
    }
  },
  'when pending to get table data': {
    type: 'GET_ENGAGEMENT_LINK_METRICS_PENDING'
  },
  'when succeeds to get table data': {
    type: 'GET_ENGAGEMENT_LINK_METRICS_SUCCESS',
    payload: [
      { count_clicked: 1802, count_raw_clicked_approx: 1692, link_name: 'Raw URL' },
      { count_clicked: 1402, count_raw_clicked_approx: 1020, link_name: 'Nintendo' }
    ]
  }
});
