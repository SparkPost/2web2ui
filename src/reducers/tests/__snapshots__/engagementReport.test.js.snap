// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Report Reducer when failed to get chart data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Array [],
    "error": Object {
      "message": "Oh no!",
    },
    "loading": false,
  },
  "linkMetrics": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
}
`;

exports[`Engagement Report Reducer when failed to get table data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Object {},
    "error": null,
    "loading": false,
  },
  "linkMetrics": Object {
    "data": Array [],
    "error": Object {
      "message": "Oh no!",
    },
    "loading": false,
  },
}
`;

exports[`Engagement Report Reducer when pending to get chart data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Object {},
    "error": null,
    "loading": true,
  },
  "linkMetrics": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
}
`;

exports[`Engagement Report Reducer when pending to get table data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Object {},
    "error": null,
    "loading": false,
  },
  "linkMetrics": Object {
    "data": Array [],
    "error": null,
    "loading": true,
  },
}
`;

exports[`Engagement Report Reducer when succeeds to get chart data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Object {
      "count_accepted": 237433,
      "count_targeted": 272234,
    },
    "error": null,
    "loading": false,
  },
  "linkMetrics": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
}
`;

exports[`Engagement Report Reducer when succeeds to get table data 1`] = `
Object {
  "aggregateMetrics": Object {
    "data": Object {},
    "error": null,
    "loading": false,
  },
  "linkMetrics": Object {
    "data": Array [
      Object {
        "count_clicked": 1802,
        "count_raw_clicked_approx": 1692,
        "link_name": "Raw URL",
      },
      Object {
        "count_clicked": 1402,
        "count_raw_clicked_approx": 1020,
        "link_name": "Nintendo",
      },
    ],
    "error": null,
    "loading": false,
  },
}
`;
